kernel/src/expressions/mod.rs

```diff
@@ -110,6 +111,50 @@ pub enum UnaryOperator {
 
 pub type ExpressionRef = std::sync::Arc<Expression>;
 
+#[derive(Clone, Debug, PartialEq)]
+pub struct UnaryExpression {
+    /// The operator.
+    pub op: UnaryOperator,
+    /// The expression.
+    pub expr: Box<Expression>,
+}
+impl UnaryExpression {
+    fn new(op: UnaryOperator, expr: impl Into<Expression>) -> Self {
+        let expr = Box::new(expr.into());
+        Self { op, expr }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct BinaryExpression {
+    /// The operator.
+    pub op: BinaryOperator,
+    /// The left-hand side of the operation.
+    pub left: Box<Expression>,
+    /// The right-hand side of the operation.
+    pub right: Box<Expression>,
```

Comment:
NOTE: While it would be tempting to do `Box<BinaryExpression>` instead of two boxes here, that would prevent match arms from unpacking it. So we keep the boxes inside the struct, and out of the way.